app/controllers/dish_controller.rb

```diff
@@ -3,4 +3,30 @@ class DishController < ApplicationController
   def show
     @dish = Dish.find(params[:id])
   end
+
+  def new
+    @menu = Menu.find(params[:menu_id])
+    @dish = Dish.new(menu: @menu)
+  end
+
+  def create
+    @menu = Menu.find(params[:menu_id])
+    @dish = Dish.new(dish_params)
+    if @dish.save
+      flash[:notice] = "A new dish has been added to #{@menu.name} menu"
+      redirect_to restaurant_menu_path(@menu.restaurant, @menu)
+    else
+      flash[:notice] = 'Please make sure to fill in all the fields'
```

Comment:
A bit to generic for my taste. I would like to see the actual errors you get with `@dish.errors`